nfp/preprocessing/xtb_preprocessor.py

```diff
@@ -0,0 +1,101 @@
+from typing import Dict, Optional, Callable, Hashable
+
+import networkx as nx
+import numpy as np
+import rdkit.Chem
+import json
+
+try:
+    import tensorflow as tf
+except ImportError:
+    tf = None
+
+from nfp.preprocessing import features
+from nfp.preprocessing.mol_preprocessor import MolPreprocessor
+from nfp.preprocessing.tokenizer import Tokenizer
+
+
+class xTBSmilesPreprocessor(MolPreprocessor):
+    def __init__(self, *args, explicit_hs: bool = True,
+        xtb_atom_features: Optional[Callable[[rdkit.Chem.Atom], Hashable]] = None,
+        xtb_bond_features: Optional[Callable[[rdkit.Chem.Bond], Hashable]] = None,
+        cutoff: float = 0.3, **kwargs):
+        super(xTBSmilesPreprocessor, self).__init__(*args, **kwargs)
+        self.explicit_hs = explicit_hs
+        self.cutoff = cutoff
+
+        #update only bond features as we dont use rdkit
+        self.bond_features = features.bond_features_wbo
+
+        if xtb_atom_features is None:
+            self.xtb_atom_features = [
+                'mulliken charges', 'cm5 charges', 's proportion', 'p proportion',
+                'd proportion', 'FOD','FOD s proportion','FOD p proportion','FOD d proportion'
+            ]
+
+        if xtb_bond_features is None:
+            self.xtb_bond_features = [
+                'Wiberg bond order'
+            ]
+
+    def create_nx_graph(self, smiles: str, jsonfile: str, **kwargs) -> nx.DiGraph:
+        mol = rdkit.Chem.MolFromSmiles(smiles)
+        with open(jsonfile, 'r') as f:
+            json_data = json.load(f)
+        if self.explicit_hs:
+            mol = rdkit.Chem.AddHs(mol)
```

Comment:
if you remove these lines (and the mol = ... line), then you could call this `xTBPreprocessor` and later create the xTBSmilesPreprocessor as I showed in another comment

Reply:
Done